### ISSUE TITLE
Reviewer Step Error: Trying to get property 'site' of non-object

### ISSUE THREAD

**Issue by riusi88** [engram-design/Workflow#145]
**Description**
Attempting to use Workflow in order to setup a 3 phased review process. Editor creates draft and submits for review >> notifies Reviewer User Group who approves >>  Publisher User Group is notified and publishes. The plugin appears to fail when an Editor submits a draft for approval if a Reviewer User Group is selected in the Workflow General Settings. The plugin works fine if we remove the Reviewer User Group, but it's critical for our project that this Review step work. 

**Steps to reproduce**
1. Create 3 User Groups [Editor, Reviewer, Publisher]
2. Setup User Group permissions based on Docs https://verbb.io/craft-plugins/workflow/docs/feature-tour/reviews. Reviewer User Group has the same permissions as an Editor. 
3. Editor Creates new Draft and submits for Review
4. Error: Trying to get property 'site' of non-object in 

**Additional info**
- Plugin version: 1.7.2
- Craft version: Craft Pro 3.7.37
- Multi-site: No

<img width="805" alt="image" src="https://user-images.githubusercontent.com/5905213/162013589-d2df99d5-e2cb-421e-aaeb-8204ef88e00e.png">

**Error Log**
yii\base\ErrorException: Trying to get property 'site' of non-object in /Users/redacted/git/website/craft3/vendor/verbb/workflow/src/services/Submissions.php:73
Stack trace:
0 /Users/redacted/git/website/craft3/vendor/craftcms/cms/src/web/ErrorHandler.php(84): craft\web\ErrorHandler->handleError()
1 /Users/redacted/git/website/craft3/vendor/verbb/workflow/src/services/Submissions.php(73): craft\web\ErrorHandler->handleError()
2 /Users/redacted/git/website/craft3/vendor/verbb/workflow/src/services/Submissions.php(340): verbb\workflow\services\Submissions->getNextReviewerUserGroup()
3 /Users/redacted/git/website/craft3/vendor/verbb/workflow/src/services/Submissions.php(204): verbb\workflow\services\Submissions->sendReviewerNotificationEmail()
4 /Users/redacted/git/website/craft3/vendor/verbb/workflow/src/services/Service.php(149): verbb\workflow\services\Submissions->approveReview()
5 /Users/redacted/git/website/craft3/vendor/yiisoft/yii2/base/Event.php(312): verbb\workflow\services\Service->onAfterSaveEntry()
6 /Users/redacted/git/website/craft3/vendor/yiisoft/yii2/base/Event.php(312): ::call_user_func:{/Users/redacted/git/website/craft3/vendor/yiisoft/yii2/base/Event.php:312}()
7 /Users/redacted/git/website/craft3/vendor/yiisoft/yii2/base/Component.php(642): yii\base\Event::trigger()
8 /Users/redacted/git/website/craft3/vendor/craftcms/cms/src/base/Element.php(4148): craft\elements\Entry->trigger()
9 /Users/redacted/git/website/craft3/vendor/craftcms/cms/src/elements/Entry.php(1680): craft\elements\Entry->afterSave()
10 /Users/redacted/git/website/craft3/vendor/craftcms/cms/src/services/Elements.php(2703): craft\elements\Entry->afterSave()
11 /Users/redacted/git/website/craft3/vendor/craftcms/cms/src/services/Elements.php(784): craft\services\Elements->_saveElementInternal()
12 /Users/redacted/git/website/craft3/vendor/craftcms/cms/src/controllers/EntryRevisionsController.php(313): craft\services\Elements->saveElement()
13 /Users/redacted/git/website/craft3/vendor/yiisoft/yii2/base/InlineAction.php(57): craft\controllers\EntryRevisionsController->actionSaveDraft()
14 /Users/redacted/git/website/craft3/vendor/yiisoft/yii2/base/InlineAction.php(57): ::call_user_func_array:{/Users/redacted/git/website/craft3/vendor/yiisoft/yii2/base/InlineAction.php:57}()
15 /Users/redacted/git/website/craft3/vendor/yiisoft/yii2/base/Controller.php(178): yii\base\InlineAction->runWithParams()
16 /Users/redacted/git/website/craft3/vendor/yiisoft/yii2/base/Module.php(552): craft\controllers\EntryRevisionsController->runAction()
17 /Users/redacted/git/website/craft3/vendor/craftcms/cms/src/web/Application.php(293): craft\web\Application->runAction()
18 /Users/redacted/git/website/craft3/vendor/verbb/workflow/src/controllers/SubmissionsController.php(95): craft\web\Application->runAction()
19 /Users/redacted/git/website/craft3/vendor/yiisoft/yii2/base/InlineAction.php(57): verbb\workflow\controllers\SubmissionsController->actionSaveDraft()
20 /Users/redacted/git/website/craft3/vendor/yiisoft/yii2/base/InlineAction.php(57): ::call_user_func_array:{/Users/redacted/git/website/craft3/vendor/yiisoft/yii2/base/InlineAction.php:57}()
21 /Users/redacted/git/website/craft3/vendor/yiisoft/yii2/base/Controller.php(178): yii\base\InlineAction->runWithParams()
22 /Users/redacted/git/website/craft3/vendor/yiisoft/yii2/base/Module.php(552): verbb\workflow\controllers\SubmissionsController->runAction()
23 /Users/redacted/git/website/craft3/vendor/craftcms/cms/src/web/Application.php(293): craft\web\Application->runAction()
24 /Users/redacted/git/website/craft3/vendor/craftcms/cms/src/web/Application.php(602): craft\web\Application->runAction()
25 /Users/redacted/git/website/craft3/vendor/craftcms/cms/src/web/Application.php(272): craft\web\Application->_processActionRequest()
26 /Users/redacted/git/website/craft3/vendor/yiisoft/yii2/base/Application.php(384): craft\web\Application->handleRequest()
27 /Users/redacted/git/website/craft3/web/index.php(12): craft\web\Application->run()
28 {main}

**Comment by engram-design** [comment]
Should be fixed for the next release. To get the fix early, change your `verbb/workflow` requirement in `composer.json` to:

```
"require": {
  "verbb/workflow": "dev-craft-3 as 1.7.2",
  "...": "..."
}
```

Then run `composer update`.

**Comment by engram-design** [comment]
Fixed in [1.7.3](https://github.com/verbb/workflow/releases/tag/1.7.3)
